component/host.go

```diff
@@ -58,5 +63,20 @@ type Host interface {
 	//
 	// GetExporters can be called by the component anytime after Component.Start() begins and
 	// until Component.Shutdown() ends.
-	GetExporters() map[DataType]map[ID]Exporter
+	GetExporters() map[DataType]map[id.ID]Exporter
+
+	// ReportComponentStatus can be used by a component to communicate its status to registered status
+	// listeners. Components can use this function to indicate that they are functioning properly,
+	// or are in an error state.
+	ReportComponentStatus(event *status.ComponentEvent)
+
+	// RegisterStatusListener allows interested components to register status listeners to receive
+	// the following events:
+	// - Events reported by via the ReportComponentStatus function,
+	// - Notifications about pipeline status changes (ready, not ready).
+	RegisterStatusListener(options ...status.ListenerOption) StatusListenerUnregisterFunc
```

Comment:
Looks like `status.Listener` is not used anywhere.

Reply:
It is used in notifications.go.